handlers/accesscontrol/access-control.go

```diff
@@ -107,6 +121,18 @@ func (ac *AccessControlHandlersCollection) isAuthorized(ctx context.Context, pla
 		jwt = payload.JWT
 	}
 
+	if _, ok := hitRecordCache.data[playbackID]; ok {
+		hitRecordCache.mux.Lock()
+
+		if len(hitRecordCache.data[playbackID].hits) >= hitRecordCache.data[playbackID].rateLimit {
+			return false, nil
```

Comment:
You wont' unlock `mux` if returning in here? right?
It's good to use `defer c.mux.Unlock()` like you did in cleanupCache() if not optimising for earliest-possible unlock

Reply:
Right, good point! I added the defer unlock and fixed the cucumber tests failing - will add a unit test in a follow up PR